File: answers/leetcode/806. Number of Lines To Write String/Solution.py
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
        result = 0
        size = 0
        for char in s:
            width = widths[ord(char) - ord('a')]
            if width + size > 100:
                result += 1
                size = width
            else:
                size += width
        return [result + 1 if size > 0 else 0, size]

File: answers/leetcode/806. Number of Lines To Write String/Solution.java
class Solution {
    public int[] numberOfLines(int[] widths, String S) {
        int[] result = new int[2];
        result[0] = S.length() > 0 ? 1 : 0;
        for (char c : S.toCharArray()) {
            if (result[1] + widths[c - 'a'] > 100) {
                result[0]++;
                result[1] = 0;
            }
            result[1] += widths[c - 'a'];
        }
        return result;
    }
}